rpc/glfs-operations.c

```diff
@@ -212,7 +213,11 @@ glusterBlockCreateEntry(struct glfs *glfs, blockCreateCli *blk, char *gbid,
     ret = -1;
     goto out;
   } else {
+#if GFAPI_VERSION741
+    ret = glfs_ftruncate(tgfd, blk->size, NULL, NULL);
```

Comment:
I think this change had been introduced with glusterfs-6, not 4.1.

Reply:
Yeah, right, I make one mistake and just checked this from the tags not the release branches, by using the 'git tag --contains SHA', but this was reverted later :-)

```
commit 248152767b0599986bbb6bb35fc27197f6be6964
Author: Kinglong Mee <kinglongmee@gmail.com>
Date:   Thu Feb 1 10:13:07 2018 +0800

    gfapi: return pre/post attributes from glfs_ftruncate
    
    Updates: #389
    Change-Id: I8faea0828921fb17f05f7321c3cb01747373f21e
    Signed-off-by: Kinglong Mee <mijinlong@open-fs.com>

diff --git a/api/src/glfs.h b/api/src/glfs.h
index 2addacb..e2f2a2c 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -571,8 +571,9 @@ off_t glfs_lseek (glfs_fd_t *fd, off_t offset, int whence) __THROW
 int glfs_truncate (glfs_t *fs, const char *path, off_t length) __THROW
         GFAPI_PUBLIC(glfs_truncate, 3.7.15);
 
-int glfs_ftruncate (glfs_fd_t *fd, off_t length) __THROW
-        GFAPI_PUBLIC(glfs_ftruncate, 3.4.0);
+int glfs_ftruncate (glfs_fd_t *fd, off_t length, struct stat *prestat,
+                    struct stat *poststat) __THROW
+        GFAPI_PUBLIC(glfs_ftruncate, 4.0.0);
 int glfs_ftruncate_async (glfs_fd_t *fd, off_t length, glfs_io_cbk fn,
                          void *data) __THROW
         GFAPI_PUBLIC(glfs_ftruncate_async, 3.4.0);

commit 09943beb499617212f2985ca8ea9ecd1ed1b470e
[root@rhel3 glusterfs]# git tag --contains 248152767b0599986bbb6bb35fc27197f6be6964
help
v4.1.0
v4.1.0alpha
v4.1.0rc0
v4.1.1
v4.1.2
v4.1.3
v4.1.4
v4.1.5
v4.1.6
v4.1.7
v4.2dev
v5.0
v5.0alpha
v5.0rc0
v5.0rc1
v5.1
v5.2
v5.3
v5.4
v6.0alpha
v6.0rc0
v6.0rc1
v6dev
v7dev
[root@rhel3 glusterfs]# 
```

Will fix it later, thanks.